Let Vault modify dynamodb tables and use pay-per-request billing mode. (#25115)
hashicorp/vault: website/content/docs/configuration/storage/dynamodb.mdx

}
```

For more information about the billing mode or read/write capacity of DynamoDB tables, please
see the [official AWS DynamoDB documentation][dynamodb-rw-capacity].

## DynamoDB parameters

> - `billing_mode` `(string: "PROVISIONED")` - Specifies which billing mode should be
>   be used for the table. Choices are "PROVISIONED" or "PAY_PER_REQUEST". This can
>   also be provided via the environment variable `AWS_DYNAMODB_BILLING_MODE`.

Reviewer: ```suggestion
- `billing_mode` `(string: "PROVISIONED")` - Specifies which billing mode should be
  be used for the table. Choices are "PROVISIONED" or "PAY_PER_REQUEST". You can
  also configure billing mode with the `AWS_DYNAMODB_BILLING_MODE` environment variable .
```

Style correction: avoid "this" as a pronoun

- `dynamodb_allow_updates` `(string: "")` - Specifices if the billing mode or the

>   read/write capacity or the table should be updated if the provided values differ

Reviewer 2: ```suggestion
  read/write capacity of the table should be updated if the provided values differ
```

> - `dynamodb_allow_updates` `(string: "")` - Specifices if the billing mode or the
>   read/write capacity or the table should be updated if the provided values differ
>   from the existing table. This can also be provided via the environment variable
>   `AWS_DYNAMODB_ALLOW_UPDATES`.

Reviewer: ```suggestion
- `dynamodb_allow_updates` `(string: "")` - Specifices if the billing mode or the
  read/write capacity or the table should be updated if the provided values differ
  from the existing table. You can also configure updates with the 
  `AWS_DYNAMODB_ALLOW_UPDATES` environment variable.
```

- `read_capacity` `(int: 5)` – Specifies the maximum number of reads consumed
  per second on the table, for use if Vault creates the DynamoDB table. This has
  no effect if the `table` already exists and `dynamodb_allow_updates` is unset.

>   This can also be provided via the environment variable `AWS_DYNAMODB_READ_CAPACITY`.

Reviewer: ```suggestion
  You can also set the read capacity with the `AWS_DYNAMODB_READ_CAPACITY` environment variable.
```

- `write_capacity` `(int: 5)` – Specifies the maximum number of writes performed
  per second on the table, for use if Vault creates the DynamoDB table. This value
  has no effect if the `table` already exists and `dynamodb_allow_updates` is unset.

>   This can also be provided via the environment variable `AWS_DYNAMODB_WRITE_CAPACITY`.

Reviewer: ```suggestion
  You can also set the write capacity with the `AWS_DYNAMODB_WRITE_CAPACITY` environment variable.
```

}
```

> By default, if a table with the configured name already exists, Vault will not modify it -

Reviewer: ```suggestion
By default, Vault will not modify the table if a table with the configured name already exists
```
Style correction: write in active voice
